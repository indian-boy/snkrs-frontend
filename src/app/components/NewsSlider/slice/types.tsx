import { FunctionComponent, SVGProps } from 'react';

export interface SliderItem {
  id: number;
  title: string;
  link: {
    label: string;
    url: string;
  };
  Image: FunctionComponent<
    SVGProps<SVGSVGElement> & { title?: string | undefined }
  >;
}

export interface NewsSliceState {
  selected: SliderItem;
  items: SliderItem[];
}
