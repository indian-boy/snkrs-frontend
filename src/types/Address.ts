export interface Address {
  latitude: number;
  longitude: number;
  location: string;
}
