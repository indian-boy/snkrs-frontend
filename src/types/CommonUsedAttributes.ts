export interface CommonUsedAttributes {
  'data-testid': string;
}
