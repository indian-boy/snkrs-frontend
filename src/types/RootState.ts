import { NewsSliceState } from 'app/components/NewsSlider/slice/types';
import { ThemeState } from 'styles/themes/slice/types';

// [IMPORT NEW CONTAINERSTATE ABOVE] < Needed for generating containers seamlessly

/*
  Because the redux-injectors injects your reducers asynchronously somewhere in your code
  You have to declare them here manually
*/
export interface RootState {
  theme?: ThemeState;
  newsSlice?: NewsSliceState;
  // [INSERT NEW REDUCER KEY ABOVE] < Needed for generating containers seamlessly
}
