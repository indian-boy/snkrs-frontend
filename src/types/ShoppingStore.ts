import { Address } from './Address';

export interface ShoppingStore {
  id: number;
  placeName: string;
  address: Address;
  availability: number;
  serviceHours: string;
}
