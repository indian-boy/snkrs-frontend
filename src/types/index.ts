import { Address } from './Address';
import { CommonUsedAttributes } from './CommonUsedAttributes';
import { RootState } from './RootState';
import { ShoppingStore } from './ShoppingStore';

export type { RootState, Address, ShoppingStore, CommonUsedAttributes };
