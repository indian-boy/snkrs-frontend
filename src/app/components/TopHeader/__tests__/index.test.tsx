import { Store } from '@reduxjs/toolkit';
import { render } from '@testing-library/react';
import * as React from 'react';
import { Provider } from 'react-redux';
import { configureAppStore } from 'store/configureStore';
import { CommonUsedAttributes } from 'types';
import { TopHeader } from '..';

jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: str => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    };
  },
}));

const renderComponentWithProviders = (
  props: Parameters<typeof TopHeader>[number] & CommonUsedAttributes,
  store: Store,
) =>
  render(
    <Provider store={store}>
      <TopHeader {...props} />
    </Provider>,
  );

describe('<TopHeader  />', () => {
  let store: ReturnType<typeof configureAppStore>;

  beforeEach(() => {
    store = configureAppStore();
  });

  it('should match snapshot', () => {
    const { getByTestId, container } = renderComponentWithProviders(
      {
        'data-testid': 'topHeaderID',
      },
      store,
    );

    const topHeaderElement = getByTestId('topHeaderID');

    expect(topHeaderElement).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
