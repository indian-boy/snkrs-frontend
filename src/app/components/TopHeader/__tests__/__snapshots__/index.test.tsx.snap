// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopHeader  /> should match snapshot 1`] = `
.c0 {
  display: none;
  background-color: #F7F7F7;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 1rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 1rem;
}

.c2 {
  width: 1.8rem;
  height: 1.8rem;
}

.c3 {
  width: 2.9rem;
  height: 2.9rem;
}

.c5 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #737373;
  font-size: 0.75rem;
  line-height: 1.5rem;
}

.c5:visited {
  color: #737373;
}

.c6 {
  height: 1.5rem;
  border-left: 1px solid #737373;
}

@media (min-width:1024px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    padding: 1rem 2.5rem;
  }
}

<div
  class="c0"
  data-testid="topHeaderID"
>
  <div
    class="c1"
  >
    <svg
      class="c2"
    >
      jordan.svg
    </svg>
    <svg
      class="c3"
    >
      snkrs.svg
    </svg>
  </div>
  <div
    class="c4"
  >
    <a
      aria-current="page"
      class="c5 active"
      href="/"
    >
      components.topHeader.links.help
      Ajuda
    </a>
    <span
      class="c6"
    />
    <a
      aria-current="page"
      class="c5 active"
      href="/"
    >
      components.topHeader.links.followYourOrder
      Acompanhe seu pedido
    </a>
    <span
      class="c6"
    />
    <a
      aria-current="page"
      class="c5 active"
      href="/"
    >
      components.topHeader.links.joinUs
      Junte-se a n√≥s
    </a>
    <span
      class="c6"
    />
    <a
      aria-current="page"
      class="c5 active"
      href="/"
    >
      components.topHeader.links.login
      Entrar
    </a>
  </div>
</div>
`;
